File: android/Dev/app/src/main/java/com/dxmnd/mos/dev/map/naver_map/n_map_util/NMapFragment.java
package com.dxmnd.mos.dev.map.naver_map.n_map_util;

import android.support.v4.app.Fragment;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.nhn.android.maps.NMapContext;
import com.nhn.android.maps.NMapView;


public class NMapFragment extends Fragment {
    private NMapContext mMapContext;

    /**
     * Fragment에 포함된 NMapView 객체를 반환함
     */
    private NMapView findMapView(View v) {

        if (!(v instanceof ViewGroup)) {
            return null;
        }

        ViewGroup vg = (ViewGroup)v;
        if (vg instanceof NMapView) {
            return (NMapView)vg;
        }

        for (int i = 0; i < vg.getChildCount(); i++) {

            View child = vg.getChildAt(i);
            if (!(child instanceof ViewGroup)) {
                continue;
            }

            NMapView mapView = findMapView(child);
            if (mapView != null) {
                return mapView;
            }
        }
        return null;
    }

    /* Fragment 라이프사이클에 따라서 NMapContext의 해당 API를 호출함 */

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mMapContext =  new NMapContext(super.getActivity());

        mMapContext.onCreate();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        throw new IllegalArgumentException("onCreateView should be implemented in the subclass of NMapFragment.");

    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        // Fragment에 포함된 NMapView 객체 찾기
        NMapView mapView = findMapView(super.getView());
        if (mapView == null) {
            throw new IllegalArgumentException("NMapFragment dose not have an instance of NMapView.");
        }

        // NMapActivity를 상속하지 않는 경우에는 NMapView 객체 생성후 반드시 setupMapView()를 호출해야함.
        mMapContext.setupMapView(mapView);
    }

    @Override
    public void onStart(){
        super.onStart();

        mMapContext.onStart();
    }

    @Override
    public void onResume() {
        super.onResume();

        mMapContext.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();

        mMapContext.onPause();
    }

    @Override
    public void onStop() {

        mMapContext.onStop();

        super.onStop();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

    @Override
    public void onDestroy() {
        mMapContext.onDestroy();

        super.onDestroy();
    }
}

File: android/RightNow/app/src/main/java/dxmnd/com/rightnow/main/info/view/NMapFragment.kt
package dxmnd.com.rightnow.main.info.view

import android.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.nhn.android.maps.NMapContext
import com.nhn.android.maps.NMapView
import dxmnd.com.rightnow.util.log

open class NMapFragment : Fragment() {
    private var mapContext : NMapContext?=null

    fun findMapView(v: View) : NMapView? {
        if (v !is ViewGroup) {
            return null
        }

        val vg : ViewGroup = v
        if(vg is NMapView) {
            return vg
        }

        for(x in 0 until vg.childCount) {
            val child: View = vg.getChildAt(x) as? ViewGroup ?: continue

            val mapView = findMapView(child)
            mapView?.let {
                return it
            }

        }
        return null
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mapContext = NMapContext(super.getActivity())
        mapContext?.onCreate()
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val mapView : NMapView? = findMapView(super.getView())
        mapView?.apply {
            mapContext?.setupMapView(this)
        }?: log(throw IllegalAccessException("map view fragment error"))
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View {

        throw IllegalArgumentException("onCreateView should be implemented in the subclass of NMapFragment.")
    }

    override fun onStart() {
        super.onStart()
        mapContext?.onStart()
    }

    override fun onResume() {
        super.onResume()
        mapContext?.onResume()
    }

    override fun onPause() {
        super.onPause()
        mapContext?.onPause()
    }

    override fun onStop() {
        mapContext?.onStop()
        super.onStop()
    }

    override fun onDestroy() {
        mapContext?.onDestroy()
        super.onDestroy()
    }

}